src/Agent.Listener/CommandLine/ConfigureAgent.cs

```diff
@@ -18,6 +18,9 @@ public class ConfigureAgent : ConfigureOrRemoveBase
         [Option(Constants.Agent.CommandLine.Flags.AddMachineGroupTags)]
         public bool AddMachineGroupTags { get; set; }
 
+        [Option(Constants.Agent.CommandLine.Flags.AllowWorkDirectoryRepositories)]
+        public bool AllowWorkDirectoryRepositories { get; set; }
```

Comment:
Is it really necessary to have this setting? Since to my understanding the (public) hosted agents will create a "fresh" agent each time, so there is no risk of having anyone else's code on you agent. If the hosted agents might still have code from another organisation on it this would be a very serious security issue.